crates/astria-sequencer/src/proposal/commitment.rs

```diff
@@ -67,6 +66,7 @@ pub(crate) fn generate_rollup_datas_commitment(
         );
     }
 
+    let rollup_ids_root = merkle::Tree::from_leaves(rollup_ids_to_txs.keys()).root();
     rollup_ids_to_txs.sort_unstable_keys();
```

Comment:
Shouldn't the sorting go above the merkle tree generation her? You are potentially inserting new rollup IDs in the previous loop `(rollup_ids_to_txs.entry(rollup_id).or_default())`, so the order of this object is not guaranteed to be stable.

We should provide a unit test to ensure this.

Reply:
yeah good point, coding at midnight got me missing. will fix!!